util/codec/codec.go

```diff
@@ -163,6 +170,13 @@ func DecodeKey(b []byte) ([]interface{}, error) {
 			}
 		case formatBytesFlag:
 			b, v[i], err = DecodeBytes(b)
+		case formatDurationFlag:
+			var r int64
+			b, r, err = DecodeInt(b)
+			if err == nil {
+				// use max fsp, let outer to do round manually.
+				v[i] = mysql.Duration{Duration: time.Duration(r), Fsp: mysql.MaxFsp}
```

Comment:
maybe no need here, but I am not sure. 
As you know, we handle encode any fsp in table value codec or time marshal too, just let outer do the convert manually.